_posts/2025-06-09-roadmap-update-html-report.md

```diff
@@ -0,0 +1,91 @@
+
+## Reqnroll HTML Report
+
+While this is still in the beta phase, I would like to share some details about Reqnroll HTML report generation. The beta phase in this case means that the code has been implemented and is working, but we are in the middle of the review process and making some small fixes.
+
+As mentioned above, our strategy was to be able to produce Cucumber Messages so that we could use existing tooling to turn that into HTML. We were aware of the [Cucumber React Components](https://github.com/cucumber/react-components) project, which provides HTML visualization for Cucumber Messages, but in the end, we found another important open-source component, [Cucumber HTML Formatter](https://github.com/cucumber/html-formatter/), which provides an end-to-end wrapper for producing a complete HTML report from Cucumber Messages. Exactly what we needed!
+
+Have you heard of "Cucumber Formatters"? That is a feature and extensibility point in Cucumber tools that can be used to produce reports, outputs, or other results in a specific format. Since Cucumber started as a command-line tool, the concept of formatters has existed there for a long time — basically, when you executed Cucumber, it was a formatter that produced the console output and another formatter that generated the test result JSON file. As SpecFlow/Reqnroll has been executed by the .NET test execution hosts (e.g., `dotnet test` or *Visual Studio Test Explorer*), we had not implemented this particular concept before.
+
+During the implementation of the Cucumber Messages output, we realized that what we were doing was essentially the same as what Cucumber calls "formatters," so we (better to say Chris) reframed our solution and implemented the formatter concept from Cucumber into Reqnroll.
+
+And if we have formatters in Reqnroll, then we can implement the Cucumber Messages output as one formatter and the HTML output as another! That means we will not need any additional tool to convert Cucumber Messages to HTML (as was needed for "SpecFlow+ LivingDoc Generator"), but it will generate the HTML file out of the box! Yes, even if you run the tests from Visual Studio!
+
+It is important to mention that all parts of the HTML generation toolchain (Reqnroll, formatters, Cucumber HTML formatter, Cucumber React Components) are free and open source.
+
+The formatter concept has two important characteristics that make it even more appealing for Reqnroll:
+1. You can register multiple formatters. If you need to have both Cucumber Messages and an HTML report, you can register both!
+2. You can write your own formatter if you need a customized report or output format! This is all integrated into the Reqnroll Plugin infrastructure.
+
+Currently, the Cucumber Messages and HTML formatters will be packaged with Reqnroll, so these can be used out of the box. Later, we might extend the set of formatters (built-in or as plugins), and I'm pretty sure there will be formatters from the community as well.
```

Comment:
The follow-on question we are likely to receive is whether users can customize the html template and css. At present the answer is no (not without significant difficulty). The html assets are bundled into the html-formatter and react-components as assembly resources. The cucumber team does not yet have an extensibility point to allow for easy customization.
Your choice whether to head-off this line of requests now or whether or not to promise anything in regards to extensibility of templates.